content/6_Advanced/Treaps.mdx

```diff
@@ -207,7 +255,267 @@ int main() {
 For speed / memory, use arrays of fixed size rather than pointers.
 </Optional>
 
+<FocusProblem problem="sam2"/>
+
+# Implicit Treaps
+
+In their most basic form, Treaps are not very useful (languages like C++
+and Java already have a built-in self-balancing binary tree that is much more
+efficient than treaps). However, with implicit treaps, we can efficiently
+perform operations on a regular array in fashion similar to segment trees and
+fenwick trees. The following operations are supported by implicit treaps:
+
+- Insert an element $x$ at position $i$
+- Delete the element at position $i$
+- Performing interval queries (sum, min, max, etc.)
+- Perform interval updates (add, set, reverse, etc.)
+
+The key behind implicit treaps lies in their name. We will use the index of the
+node to be its key. Because maintaining this value explicitely will result in up
+to $\mathcal{O}(n)$ values to be updated per insertion/deletion, we will keep
+this value implicitly.
+
+The index of a node is equal to the number of nodes less than it. It is
+important to note that these nodes can occur both in the left subtree of the
+current node as well as the node's ancestors and the left subtree of its
+ancestors. In other words,
+
+$$\texttt{index}(u) = |\{1 + \texttt{size}(P.\texttt{left}) \mid P \text{ is on path from } u \text{ to root}\}|$$
```

Comment:
Use

```
$$
equation
$$
```

to center an equation. Also, this isn't quite correct because you need to consider its children too.